### PR TITLE
PHP 8.3 support added

### PR DESCRIPTION
**Pull request by glo71317**
|    Q          |   A
|-------------- | ------
| Documentation | yes/no
| Bugfix        | yes/no
| BC Break      | yes/no
| New Feature   | yes/no
| RFC           | yes/no
| QA            | yes/no

### CODE REVIEW

**Comment by glo71317** [comment]
@froschdesign Can you please review the PR if all good please merge this PR?

File: src/MailHide.php [hunk 1]
```diff
@@ -280,7 +280,7 @@ public function setEmail($email)
         }
 
         $this->emailLocalPart  = $emailParts[0];
-        $this->emailDomainPart = $emailParts[1];
+        $this->emailDomainPart = $emailParts[1] ?? '';
```
Comment (Ocramius) [inline src/MailHide.php]: This change seems wrong: the second key must always exist.

If this was done for static analysis, I suggest reverting and adding to baseline instead.

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: src/MailHide.php (hunk 1 above)
Comment (glo17680) [reply src/MailHide.php]: Hi @Ocramius,

As you per your suggestion, we're planning to change the statement. So, the fix (as given in [Psalm](https://psalm.dev/docs/running_psalm/issues/PossiblyUndefinedArrayOffset/) ) could possibly be like:

From:
```
        $this->emailDomainPart = $emailParts[1] ?? '';
```
To:
```
        if(!isset($emailParts[1])) {
            $emailParts[1] = "";
        }

        $this->emailLocalPart  = $emailParts[0];
        $this->emailDomainPart = $emailParts[1];
```

For the below error of  QA Checks (Psalm:undefined array keys) :
```
PossiblyUndefinedArrayOffset
src/MailHide.php:283:34: PossiblyUndefinedArrayOffset: Possibly undefined array key $emailParts[1]
```

Could you please review and provide your inputs on the same ?

Comment (Ocramius) [reply src/MailHide.php]: @glo17680 I suggested expanding the baseline with no code changes to the original source.

Comment (glo17680) [reply src/MailHide.php]: We have made the suggested changes.

**Comment by glo71317** [comment]
@Ocramius @gsteel All PR review comments addressed. Hence, PR is ready more than a week, No one PR is reviewing?

**Comment by Ocramius** [comment]
Thanks @glo71317 

Squashed to avoid cluttering `git blame` with multiple internal reverted changes.

**Pull request merged**